From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Charles Kerr <charles@charleskerr.com>
Date: Sat, 16 May 2026 17:12:29 -0500
Subject: build: preserve N-API symbols under ThinLTO

Solve a build-time error where thinLTO stripped out N-API entry points because
they're only reachable via dynamic loading. We avoid this by copying the
`__attribute__((used))` approach from https://github.com/nodejs/node/pull/49037

diff --git a/src/js_native_api.h b/src/js_native_api.h
index 31eee1234e1d2ff281068837443164b7970aecf4..714c3a3c50d2822089b1744d28d31e004c820fc6 100644
--- a/src/js_native_api.h
+++ b/src/js_native_api.h
@@ -17,7 +17,7 @@
   __attribute__((visibility("default")))                                       \
   __attribute__((__import_module__("napi")))
 #else
-#define NAPI_EXTERN __attribute__((visibility("default")))
+#define NAPI_EXTERN __attribute__((visibility("default"))) __attribute__((used))
 #endif
 #endif
 
